app/models/availability.rb

```diff
@@ -1,6 +1,8 @@
 class Availability < ActiveRecord::Base
   belongs_to :person
 
+  before_create :cancel_duplicates
```

Comment:
I think this would be better as an after save hook. We would want it on save, so that if someone updated an availability to overlap, we'd still catch that. And an after hook, so that if the availability _doesn't_ save for some reason, we won't have cancelled availabilities we should not have.